src/server/game/Spells/Auras/SpellAuras.cpp

```diff
@@ -851,6 +854,38 @@ void Aura::Update(uint32 diff, Unit* caster)
             }
         }
     }
+    
+    if (m_heartBeatResistTimer)
+    {
+        if (m_heartBeatResistTimer > diff)
+            m_heartBeatResistTimer -= diff;
+        else
+        {     
+            m_heartBeatResistTimer = CalculatePct(m_maxDuration, 25);
```

Comment:
I disagree with this, it's inaccurate.

Is there any reason the expiry check shouldn't happen multiple times on the same tick, if diff is sufficiently high?

e.g.

```cpp
while (m_heartBeatResistTimer <= diff)
{
  // do the check
  diff -= m_heartBeatResistTimer;
  m_heartBeastResistTimer = CalculatePct(m_maxDuration, 25);
}
m_heartBeatResistTimer -= diff;
```

Reply:
@Eliminationzx